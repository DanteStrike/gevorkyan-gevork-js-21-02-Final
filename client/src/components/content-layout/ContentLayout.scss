@import "src/assets/variables";
@import "src/assets/mixins";

.content-layout {
  display: flex;
  flex-direction: column;
  justify-content: center;

  &--full {
    width: 100%;
    height: 100%;
    max-height: 800px;
  }

  &--content {
    align-items: center;
    width: 77%;
    min-height: 520px;
    padding: 16px;
    background-color: var(--primary-background);
    border: var(--primary-border);
    border-radius: 15px;
  }

  &__title {
    margin-bottom: 14px;
    font-weight: 700;
    color: var(--primary-font-color);
    text-align: center;
  }

  &__title--hide {
    @include visually-hidden;
  }

  &__error {
    position: fixed;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 999;
    text-align: center;
  }
}

@media (max-width: $desktopWidth) {
  .content-layout {
    &--full {
      max-height: unset;
    }

    &--content {
      width: 95%;
    }
  }
}
