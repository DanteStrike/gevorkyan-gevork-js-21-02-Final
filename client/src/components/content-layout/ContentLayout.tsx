import React from 'react';
import './ContentLayout.scss';
import {Alert} from 'antd';

export enum ContentLayoutType {
  FULL = `full`,
  CONTENT = `content`,
}

interface ContentLayoutProps {
  type: ContentLayoutType;
  title: string;
  isError?: boolean;
  errMsg?: string | null;
  hideTitle?: boolean;
  children?: React.ReactNode;
}

function ContentLayout({type, hideTitle, title, children, isError, errMsg}: ContentLayoutProps) {
  return (
    <main className={`content-layout content-layout--${type}`}>
      <h1 className={`content-layout__title ${hideTitle ? `content-layout__title--hide` : ``}`.trim()}>{title}</h1>
      {children}
      {isError && <Alert message={errMsg} type="error" className="content-layout__error" closable />}
    </main>
  );
}

ContentLayout.defaultProps = {
  hideTitle: false,
  isError: false,
  errMsg: ``,
  children: ``,
};

export default React.memo(ContentLayout);
