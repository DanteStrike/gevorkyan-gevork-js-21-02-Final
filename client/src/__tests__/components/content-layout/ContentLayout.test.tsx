import {shallow} from 'enzyme';
import React from 'react';
import ContentLayout, {ContentLayoutType} from '../../../components/content-layout/ContentLayout';

describe(`Component CustomLink should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(
      <ContentLayout title="header" type={ContentLayoutType.FULL}>
        <div id="content" />
      </ContentLayout>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot mode test`, () => {
    const wrapper = shallow(
      <ContentLayout title="header" type={ContentLayoutType.CONTENT}>
        <div id="content" />
      </ContentLayout>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot error test`, () => {
    const wrapper = shallow(
      <ContentLayout title="header" type={ContentLayoutType.CONTENT} isError errMsg="error">
        <div id="content" />
      </ContentLayout>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe(`Component CustomLink should work correctly`, () => {
  it(`snapshot error test`, () => {
    const wrapper = shallow(
      <ContentLayout title="header" type={ContentLayoutType.CONTENT}>
        <div id="content" />
      </ContentLayout>
    );
    expect(wrapper.find(`.content-layout__title.content-layout__title--hide`)).toHaveLength(0);
    wrapper.setProps({hideTitle: true});
    expect(wrapper.find(`.content-layout__title.content-layout__title--hide`)).toHaveLength(1);
  });
});
