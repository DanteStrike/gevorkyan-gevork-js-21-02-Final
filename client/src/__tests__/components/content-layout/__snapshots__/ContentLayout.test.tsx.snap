// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CustomLink should render correctly snapshot default test 1`] = `
<main
  className="content-layout content-layout--full"
>
  <h1
    className="content-layout__title"
  >
    header
  </h1>
  <div
    id="content"
  />
</main>
`;

exports[`Component CustomLink should render correctly snapshot error test 1`] = `
<main
  className="content-layout content-layout--content"
>
  <h1
    className="content-layout__title"
  >
    header
  </h1>
  <div
    id="content"
  />
  <Alert
    className="content-layout__error"
    closable={true}
    message="error"
    type="error"
  />
</main>
`;

exports[`Component CustomLink should render correctly snapshot mode test 1`] = `
<main
  className="content-layout content-layout--content"
>
  <h1
    className="content-layout__title"
  >
    header
  </h1>
  <div
    id="content"
  />
</main>
`;
